build_extensions.py

```diff
@@ -0,0 +1,118 @@
+import argparse
+import os
+import pathlib
+from subprocess import check_call
+from typing import List
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(prog="build_extensions")
+
+    parser.add_argument(
+        "--build_dir",
+        default="build",
+        help="Directory for Meson to use while building extensions.",
+    )
+    parser.add_argument(
+        "--build_type",
+        default="release",
+        choices=["debug", "debugoptimized", "release"],
+        help="The type of build to provide. Defaults to release mode.",
+    )
+    parser.add_argument(
+        "--problem",
+        default="vrptw",
+        choices=["cvrp", "vrptw"],
+        help="Which type of solver to compile. Defaults to 'vrptw'.",
+    )
+    parser.add_argument(
+        "--clean",
+        action="store_true",
+        help="Clean build and installation directories before building.",
+    )
+    parser.add_argument(
+        "--regenerate_type_stubs",
+        action="store_true",
+        help="""
+        Whether to regenerate the MyPy type stubs as well. Default False, since
+        this can overwrite manual improvements and docstrings.
+        """,
+    )
+    parser.add_argument(
+        "--additional",
+        nargs=argparse.REMAINDER,
+        default=[],
+        help="Extra Meson configuration options (passed verbatim to Meson).",
+    )
+
+    return parser.parse_args()
+
+
+def clean(build_dir: pathlib.Path, install_dir: pathlib.Path):
+    check_call(["rm", "-rf", str(build_dir)])
+
+    for extension in install_dir.rglob("*.so"):
+        extension.unlink()
+
+    for extension in install_dir.rglob("*.pyd"):
+        extension.unlink()
+
+
+def regen_stubs(install_dir: pathlib.Path):
```

Comment:
Maybe just call it `regenerate_stubs`? Also, it feels a bit weird to have this in here as it changes files under version control, which I would not per se expect in a build script, but I think it is not that important.